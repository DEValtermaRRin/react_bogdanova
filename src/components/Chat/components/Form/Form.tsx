import React, { useState, memo } from 'react';
import { Button } from './components/Button/Button';
import { Message } from './components/Message/Message';
import './Form.scss';

interface FormProps {
  addMessages: (val: string, user: string) => void;
  onSubmit?: () => void;
  userName: string;
}

export const Form = memo<FormProps>(({ userName, addMessages }) => {
  const name = 'Send message';
  const [value, setValue] = useState('');

  const handleClickSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    addMessages(value, userName);
    setValue('');
  };
  return (
    <form data-testid="form" onSubmit={handleClickSubmit} className="form">
      <Message value={value} setValue={setValue} />
      <Button disabled={!userName} name={name} />
    </form>
  );
});
