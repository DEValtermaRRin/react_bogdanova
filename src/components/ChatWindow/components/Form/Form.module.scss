.form {
  margin-top: 5px;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  height: 40px;
  width: 560px;
  &__flex {
    display: flex;
    align-items: center;
  }
}

.form__flexAround {
  display: flex;
  justify-content: space-around;
  &__flex {
    display: flex;
    align-items: center;
  }
}