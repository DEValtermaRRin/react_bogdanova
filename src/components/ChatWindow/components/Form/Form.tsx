import React, { useState, memo } from 'react';
import { useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';

import { ButtonSend } from './components/ButtonSend';
import { ButtonDel } from './components/ButtonDel';
import { Message } from './components/Message';

import { addMessage } from 'src/store/chatlist/slice';

import style from './Form.module.scss';

export interface FormProps {
  userName: string;
}

export const Form = memo<FormProps>(({ userName }) => {
  const [value, setValue] = useState('');
  const dispatch = useDispatch();
  const { chatId } = useParams();

  const handleClickSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    if (chatId && value) {
      dispatch(
        addMessage({ chatId, message: { text: value, author: userName } }),
      );
    }
    setValue('');
  };
  return (
    <form
      data-testid="form"
      onSubmit={handleClickSubmit}
      className={style.form}
    >
      <ButtonDel />
      <Message value={value} setValue={setValue} />
      <ButtonSend disabled={!userName} />
    </form>
  );
});
