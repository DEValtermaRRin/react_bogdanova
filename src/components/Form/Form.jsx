import React, { useState } from 'react';
import { Button } from './components/Button/Button';
import { Message } from './components/Message/Message';
import { User } from './components/User/User';

export const Form = ({ addMessages }) => {
  const [name] = useState('Send message');
  const [value, setValue] = useState('');
  const [userName, setUserName] = useState('');

  const handleClickSubmit = (e) => {
    e.preventDefault();
    addMessages(value, userName);
    setValue('');
  };

  return (
    <form onSubmit={handleClickSubmit} className='container'>
      <User name={userName} getName={setUserName} />
      <Message value={value} setValue={setValue} />
      <Button disabled={!userName} name={name} />
    </form>
  );
};
