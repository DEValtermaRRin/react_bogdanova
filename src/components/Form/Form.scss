.form {
  &__flex {
    display: flex;
    
  }
}