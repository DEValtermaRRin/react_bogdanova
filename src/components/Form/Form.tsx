import React, { useState, FC, memo } from 'react';
import { Button } from './components/Button/Button';
import { Message } from './components/Message/Message';
import { User } from './components/User/User';

interface FormProps {
  addMessages: (val: string, user: string) => void;
}

export const Form = memo<FormProps>(({ addMessages }) => {
  const [name] = useState('Send message');
  const [value, setValue] = useState('');
  const [userName, setUserName] = useState('');

  const handleClickSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    addMessages(value, userName);
    setValue('');
  };

  return (
    <form onSubmit={handleClickSubmit} className="container">
      <User name={userName} getName={setUserName} />
      <Message value={value} setValue={setValue} />
      <Button disabled={!userName} name={name} />
    </form>
  );
});
