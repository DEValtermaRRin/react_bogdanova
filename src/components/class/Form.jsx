import React, { Component } from 'react';
import { Input } from './Input';
import { Button } from './Button';
import { Message } from './Message';
import { ButtonClear } from './ButtonClear';

export class Form extends Component {
  state = {
    name: 'Send message',
    value: '',
    messages: [],
    btnClear: 'Clear messages',
  };

  handleClick = () => {
    this.setState({ messages: [...this.state.messages, this.state.value] });
    this.setState({ value: '' });
  };

  handleChange = (e) => {
    this.setState({ value: e.target.value });
  };

  clearMsg = () => {
    this.setState({ messages: [] });
  };

  render() {
    return (
      <>
        <Input change={this.handleChange} value={this.state.value} />
        <Button name={this.state.name} click={this.handleClick} />
        <Message send={this.state.messages} />
        <ButtonClear name={this.state.btnClear} click={this.clearMsg} />
      </>
    );
  }
}
