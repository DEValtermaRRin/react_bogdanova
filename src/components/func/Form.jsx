import React, { useState } from "react";

export const Form = ({ addMessages }) => {
  const [name] = useState("Send message");
  const [value, setValue] = useState("");
  const [userName, setUserName] = useState("");

  const handleClickSubmit = (e) => {
    e.preventDefault();
    addMessages(value, userName);
    setValue("");

  };

  return (
    <form onSubmit={handleClickSubmit} className="container">
      <input
        value={userName}
        onChange={(e) => setUserName(e.target.value)}
        type="text"
				placeholder="Your name? "
      />
      <textarea 
				value={value} 
				onChange={(e) => setValue(e.target.value)}
				placeholder="Your message" />
      <button type="submit" className="btn btn-top">
        {name}
      </button>
    </form>
  );
};
