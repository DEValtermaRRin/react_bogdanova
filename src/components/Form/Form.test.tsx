import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Form } from './Form';




describe('Form', () => {
  it('render component', () => {
    render(<Form addMessages={Object} />)
  })
  it('render with snapshot', () => {
    expect(
      render(<Form addMessages={Object} />)).toMatchSnapshot();
  });
  it('render message with text', () => {
    render(<Form  addMessages={Object}/>)
    fireEvent.input(screen.getByTestId('message'), {
      target: {value: 'Hello'}
    })
    fireEvent.click(screen.getByRole('button'))
    expect(screen.getByText(/hello/i)).toBeInTheDocument()
  })
/*   it('render user with text', () => {
    render(<Form  addMessages={Object}/>)
    fireEvent.input(screen.getByTestId('user'), {
      target: {value: 'Vasia'}
    })
    fireEvent.click(screen.getByRole('button'))
    expect(screen.getAllByText(/Vasia/)).toBeInTheDocument()
  }) */
})

