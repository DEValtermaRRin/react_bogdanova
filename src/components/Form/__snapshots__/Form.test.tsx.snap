// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form render with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="container"
      >
        <input
          data-testid="user"
          placeholder="Your name?"
          type="text"
          value=""
        />
        <textarea
          data-testid="message"
          placeholder="Your message"
        />
        <button
          class="btn btn-top"
          disabled=""
          type="submit"
        >
          Send message
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="container"
    >
      <input
        data-testid="user"
        placeholder="Your name?"
        type="text"
        value=""
      />
      <textarea
        data-testid="message"
        placeholder="Your message"
      />
      <button
        class="btn btn-top"
        disabled=""
        type="submit"
      >
        Send message
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
